### ISSUE TITLE
CarouselView: HorizontalOptions FillAndExpand has no effect (since XF 4.0 pre-8 and 9)

### ISSUE THREAD

**Issue by jckruse** [xamarin/Xamarin.Forms#6005]
### Description

While testing CarouselView in XF 4.0, I noticed that the HorizontalOptions property seems to have no effect.
Until pre-7 it works, with pre-8 and 9 the problem occurs.
The CarouselView is in a grid (7 rows, 1 column with ColumnDefinition Width="*")
The items are own controls (views) see e.g. Section 2 (Sample View) of the source code

### Steps to Reproduce

CarouselView:

```
<CarouselView Grid.Row="6"
              x:Name="Carousel"
              HorizontalOptions="FillAndExpand"
              VerticalOptions="FillAndExpand"
              ItemsSource="{Binding CarouselListItems, Mode=OneWay}"
              IsVisible="{Binding ShowCarouselView}">
  <CarouselView.ItemsLayout>
    <GridItemsLayout Orientation="Horizontal"
                     SnapPointsAlignment="Center"
                     SnapPointsType="Mandatory">
    </GridItemsLayout>
  </CarouselView.ItemsLayout>
  <CarouselView.ItemTemplate>
    <DataTemplate>
      <ContentView Content="{Binding .}" />
    </DataTemplate>
  </CarouselView.ItemTemplate>
</CarouselView>
```

Sample View:

  ```
<ContentView.Content>
    <Grid RowSpacing="0">
      <Grid.RowDefinitions>
        <RowDefinition Height="40" />
        <RowDefinition Height="*" />
      </Grid.RowDefinitions>
      <StackLayout Grid.Row="0"
                   VerticalOptions="FillAndExpand"
                   HorizontalOptions="FillAndExpand"
                   StyleClass="headerBox">
        <Label Text="aktueller Standort"
               VerticalOptions="CenterAndExpand"
               VerticalTextAlignment="Center"
               StyleClass="itemLabelStyleWhite">

        </Label>
      </StackLayout>
      <AbsoluteLayout Grid.Row="1">
        <maps:Map x:Name="Map"
                  IsShowingUser="True"
                  HasScrollEnabled="False"
                  HasZoomEnabled="False"
                  MapType="Street"
                  AbsoluteLayout.LayoutBounds="0,0,1,1"
                  AbsoluteLayout.LayoutFlags="All">
          
        </maps:Map>
      </AbsoluteLayout>
    </Grid>
  </ContentView.Content>
```

### Expected Behavior
The same behavior as before pre-7, that the CarouselView takes in the entire width of the space.
See screenshots at the end.

### Basic Information

- Version with issue: 4.0.0.346134-pre9
- Last known good version: 4.0.0.250467-pre7
- IDE: Visual Studio Community 2019 (16.0.2)
- Platform Target Frameworks: 
  - iOS:  
  - Android: 8.1 (Oreo) 
  - UWP:   
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

### Screenshots

Working with pre-7:

![Screenshot_20190418-110309](https://user-images.githubusercontent.com/7601926/56712202-ed10c800-672d-11e9-95c9-3d57bea0339b.png)

Not working with pre-8 and 9:

![Screenshot_20190418-105132](https://user-images.githubusercontent.com/7601926/56712221-0285f200-672e-11e9-9677-ae97973c56a3.png)

**Comment by rmarinho** [comment]
can you please add a small reproduction sample to be easier for us to understand?!

Thanks

**Comment by rmarinho** [comment]
This should be fixed on 5.0.0
